kpe/src/tokenizer/encoding.rs

```diff
@@ -9,34 +9,82 @@ use rubert_tokenizer::{Encoding as BertEncoding, Offsets};
 /// The token ids of the encoded sequence.
 ///
 /// The token ids are of shape `(1, token_size)`.
-#[derive(Clone, Deref, From)]
+#[derive(Clone, Debug, Deref, From)]
 pub struct TokenIds(pub Array2<i64>);
 
+impl TokenIds {
+    /// Checks if the token ids are valid, i.e. in the interval `[0, vocab_size)`.
+    pub fn is_valid(&self, vocab_size: usize) -> bool {
+        debug_assert!(vocab_size as u64 <= i64::MAX as u64);
+        self.iter()
+            .copied()
+            .all(|token_id| 0 <= token_id && token_id < vocab_size as i64)
+    }
+}
+
 /// The attention mask of the encoded sequence.
 ///
 /// The attention mask is of shape `(1, token_size)`.
-#[derive(Clone, Deref, From)]
+#[derive(Clone, Debug, Deref, From)]
 pub struct AttentionMask(pub Array2<i64>);
 
+impl AttentionMask {
+    /// Checks if the attention mask is valid, i.e. either `0` or `1`.
+    pub fn is_valid(&self) -> bool {
+        self.iter()
+            .copied()
+            .all(|attention| attention == 0 || attention == 1)
+    }
+}
+
 /// The type ids of the encoded sequence.
 ///
 /// The type ids are of shape `(1, token_size)`.
-#[derive(Clone, Deref, From)]
+#[derive(Clone, Debug, Deref, From)]
 pub struct TypeIds(pub Array2<i64>);
 
+impl TypeIds {
+    /// Checks if the type ids are valid, i.e. `0`.
+    pub fn is_valid(&self) -> bool {
+        self.iter().copied().all(|type_id| type_id == 0)
+    }
+}
+
 /// The starting tokens mask of the encoded sequence.
 ///
 /// The valid mask is of shape `(token_size,)`.
-#[derive(Clone, Deref, From)]
+#[derive(Clone, Debug, Deref, From)]
 pub struct ValidMask(pub Vec<bool>);
 
+impl ValidMask {
+    /// Gets the number of valid entries in the mask.
+    pub fn size(&self) -> usize {
```

Comment:
I would rename this `size` is often used in a similar way as length.

Maybe a name like `count_valid` or similar would be a bit more clear.

Reply:
`size` is intentional because it is indeed the size of the valid embeddings array, but i see that this could be confused with the length of the valid mask itself, so i'll rename it.